strawberry/experimental/pydantic/object_type.py

```diff
@@ -131,28 +138,18 @@ def wrap(cls):
 
         all_model_fields.extend(
             (
-                (
-                    field.name,
-                    field.type,
-                    field,
+                DataclassCreationFields(
+                    name=field.name,
+                    type_annotation=field.type,
+                    field=field,
                 )
                 for field in extra_fields + private_fields
                 if field.type != strawberry.auto
             )
         )
 
         # Sort fields so that fields with missing defaults go first
-        # because dataclasses require that fields with no defaults are defined
-        # first
-        missing_default = []
-        has_default = []
-        for field in all_model_fields:
-            if field[2].default is dataclasses.MISSING:
-                missing_default.append(field)
-            else:
-                has_default.append(field)
-
-        sorted_fields = missing_default + has_default
+        sorted_fields = sort_creation_fields(all_model_fields)
```

Comment:
this might change in future https://github.com/strawberry-graphql/strawberry/pull/1187

adding a comment so we can get back to it in the other PR, since it is still open :)